ArduCopter/mode_loiter.cpp

```diff
@@ -106,6 +106,17 @@ void ModeLoiter::run()
 
         // get pilot desired climb rate
         target_climb_rate = get_pilot_desired_climb_rate(channel_throttle->get_control_in());
+
+        if (is_negative(target_climb_rate) && copter.rangefinder_alt_ok()) {
+            const float curr_alt = copter.rangefinder_state.alt_cm_glitch_protected * 0.01; // in meters
```

Comment:
I think we should use the glitch protected and filtered rangefinder altitude, rangefinder_state.alt_cm_filt.

Reply:
Thanks. I have fixed this.